### PR TITLE
fix: Installer overall timeout to None

### LINKED ISSUE #22120: solana installer times out while installing
#### Problem
Build in CI times out while downloading the installer binaries

Installer seems to be ~250 MB and is given at most 30 seconds according to the reqwest client default.

This leads to this error while building in CI
```
InstalledDir: /usr/bin
downloading v1.9.2 installer
Error: error decoding response body: operation timed out
Error: Process completed with exit code 1.
```

#### Proposed Solution

Override the timeout to something more reasonable that the 30 seconds default, possibly a few minutes at least?
https://github.com/solana-labs/solana/blob/master/install/src/command.rs#L94-L107

Does that sound correct or is this symptom hiding something else?

### PR DESCRIPTION
**Pull request by Arrowana**
#### Summary of Changes

Fixes #22120

### CODE REVIEW

**Comment by mvines** [comment]
I'd be game for an even longer timeout, like 10 minutes maybe.  But I don't feel strongly

**Comment by Arrowana** [comment]
What if `None`? Then it will hang, letting whatever application timeout on top kill it?

**Comment by mvines** [comment]
I'm fine with None too.  I assumed there was no timeout anyway :)

**Comment by t-nelson** [comment]
I'm also fine with no timeout for read/write ops.  I think we should specify `connect_timeout()` in its place though

**Comment by Arrowana** [comment]
@mvines @t-nelson timeout to None and connect_timeout to 5 seconds.

File: install/src/command.rs [hunk 1]
```diff
@@ -91,7 +91,10 @@ fn download_to_temp(
     let temp_dir = TempDir::new()?;
     let temp_file = temp_dir.path().join("download");
 
-    let client = reqwest::blocking::Client::new();
+    let client = reqwest::blocking::Client::builder()
+        .connect_timeout(Duration::from_secs(5))
```
Comment (t-nelson) [inline install/src/command.rs:95]:
```suggestion
        .connect_timeout(Duration::from_secs(30))
```

Thanks!  Let's go with 30s.  Someone's bound to try this on WWAN from the sticks

**Review by mvines — APPROVED** [review]

**Pull request merged**